classes/record/RecordDAO.inc.php

```diff
@@ -264,13 +264,28 @@ function getInsertRecordId() {
 	/**
 	 * Retrieve a count of the records available.
 	 * @param $archiveId Specific archive to query (optional)
+	 * @param $onlyEnabled boolean
 	 * @return int
 	 */
-	function getRecordCount($archiveId = null) {
-		$result =& $this->retrieve(
-			'SELECT COUNT(*) AS count FROM records' . (isset($archiveId)?' WHERE archive_id = ?':''),
-			isset($archiveId)?$archiveId:false
-		);
+	function getRecordCount($archiveId = null, $onlyEnabled = false) {
+		$sql = 'SELECT COUNT(*) AS count
+				FROM records r';
+
+		if ($onlyEnabled) {
```

Comment:
I'm thinking something like...
````
$result =& $this->retrieve(
    SELECT COUNT(*) AS count
    FROM records r 
    ' . ($onlyEnabled?', archives a ':'') . '
    WHERE 1=1
    ' . ($archiveId?' AND r.archive_id = ? ':'') . '
    ' . ($onlyEnabled?' AND r.archive_id = a.archive_id ':'') . '
    $archiveId? (int) $archiveId:false
);
````
(Untested -- just for the gist. But this kind of thing helps keep DAOs short and maintainable.)